wasm/src/avs_pc.ts

```diff
@@ -1026,7 +1028,37 @@ function update_tracks(pc: PeerConnection, stream: MediaStream): Promise<void> {
     }).then();
 }
 
-function getEncodingParameter(sender: RTCRtpSender, isScreenShare: boolean) {
+type DownScaleFactor = { scaleResolutionDownBy: number, maxBitrate: number };
+
+function determineDownScaleFactor(videoTrack: MediaStreamTrack, isScreenShare: boolean) {
+    // Firefox do not support MediaTrackSettings for screen share tracks, so we ignore this one.
+    if (!isScreenShare || pc_env === ENV_FIREFOX) {
+        return {scaleResolutionDownBy: 2.0, maxBitrate: 800000}
+    }
+
+    // do screen share downscaling
+    const settings: MediaTrackSettings = videoTrack.getSettings();
+    // No downscaling for SD
+    if (settings.width <= 640 || settings.height <= 480) {
+        return {scaleResolutionDownBy: 1.0, maxBitrate: 300000}
+    }
+    // Down scale HD (should be the most case)
+    if (settings.width <= 1280 || settings.height <= 720) {
+        return {scaleResolutionDownBy: 1.3, maxBitrate: 80000}
```

Comment:
This is what we discussed in the chat, maybe better to do 1.5 downscale, as this brings the 720p to 480p, also maxBitrate seems too low, a guess just a typo...